src/deserialize.js

```diff
@@ -32,9 +32,12 @@ const deserializeRelationships = (relationships, included) => (
       deserializedResource = deserializeResources(data, included, links)
     } else if (isPlainObject(data)) {
       deserializedResource = deserializeResource(data, included, links)
+      if (isEmpty(deserializedResource)) deserializedResource = null
+    } else if (data == null) {
+      deserializedResource = null
```

Comment:
Wdyt about this? Doesn't change the behavior for valid JSON API, but is more fault tolerant.

```suggestion
      if (isEmpty(deserializedResource)) deserializedResource = null
    } else {
      deserializedResource = null
```

Reply:
I think removing the isEmpty line is not a good idea because it permits an empty object to be set (which would break some checks in our apps that rely on presence/absence). I am fine with removing the condition for the else case, though.

Reply:
But the diff is only the removal of the `if` clause 🤔 😄